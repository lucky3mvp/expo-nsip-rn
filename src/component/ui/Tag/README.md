---
title: Tag
subTitle: 标签
---
# 标签
### 组件描述
- 可选择，可关闭的标签元素
### 示例代码
```html
<Tag>普通Tag</Tag>
<Tag defaultSelected>默认已选Tag</Tag>
<Tag disabled>禁用Tag</Tag>
<Tag onClose={()=>doSomething()}>禁用Tag</Tag>
<Tag onChange={()=>doSomething()}>禁用Tag</Tag>
```

## API

### props列表

属性 | 说明 | 类型 | 默认值
----|-----|------|------
| type | 主题类型 | string | default |
| size | 尺寸 | string | 可选择small,large |
| text | 标签显示的文本 | - | - |
| children | 标签显示的文本，使用时 text 属性失效 | - | - |
| selected | 是否选择 | boolean | false |
| defaultSelected | 是否默认选择 | boolean | false |
| disabled | 是否禁用 | boolean | false |
| closed | 是否可以关闭 | boolean | false |
| onClose | 关闭后的回调 | func | |
| onChange | 点击后的回调 | func | |
| closeIconName | 关闭按钮图标名称 | string | |
| closeIconFamily | 关闭按钮图标字体库 | string | |

### 样式对象styles

属性 | 说明 | 适用类型
----|-----|------
| container | Tag容器样式 | View |
| text | Tag文本样式 | Text |
| textSelected | 已选Tag文本样式 | Text |
| textDisabled | 禁用Tag文本样式 | Text |
