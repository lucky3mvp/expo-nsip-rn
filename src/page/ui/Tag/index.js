import React, { Component } from 'react'
import { ScrollView, Text, View, StyleSheet } from 'react-native'
import { Style } from 'common'
import { Tag, Icon } from 'component'

const style = StyleSheet.create({
	container: {
		flex: 1,
		margin: 10,
	},

	header: {
		marginLeft: 10,
		marginTop: 10
	},

	verticalContainer: {
		flexDirection: 'column',
		alignItems: 'flex-start'
	},

	horizonContainer: {
		flexDirection: 'row',
		alignItems: 'flex-end'
	}

})

export default class Page extends Component {
	state = {
		selected1: true,
		selfTagSel: false,
		closed1: false
	}
	constructor(props) {
		super(props)
		Tag.setBaseStyle(Tag, {
			container: {
				marginLeft: 10,
				marginTop: 5,
				marginBottom: 5
			}
		})
	}

	render() {
		return (
			<ScrollView style={Style.container}>
				<View style={style.header}>
					<Text>纵向布局</Text>
				</View>
				<View style={style.verticalContainer}>
					<Tag text='使用 text props' />
					<Tag text='text属性' >children属性，text属性失效</Tag>
					<Tag >默认为非控组件</Tag>
					<Tag selected={this.state.selected1} onChange={
						selected => this.setState({ selected1: selected })
					} >受控属性：selected</Tag>

					<Tag disabled>禁止的</Tag>

					<Tag closed={this.state.closed1} onClose={
						closed => this.setState({ closed1: closed })
					}>受控属性：closed，开启关闭功能</Tag>

					<Tag styles={{
						selected: { borderColor: '#891232' }
					}} selected={this.state.selfTagSel} onChange={
						selected => this.setState({ selfTagSel: selected })
					}>
						<View style={{ flexDirection: 'row' }}>
							<Icon style={{ color: this.state.selfTagSel ? '#891232' : '#bbb' }} name="calendar-check-o" />
							<Text style={{ color: this.state.selfTagSel ? '#891232' : '#bbb' }}>自定义样式和内容</Text>
							<Icon style={{ color: this.state.selfTagSel ? '#891232' : '#bbb' }} name="check" />
						</View>
					</Tag>


					<Tag size="small">small</Tag>
					<Tag size="large">large</Tag>
					<Tag size="small" closed={false}>small</Tag>
					<Tag size="large" closed={false}>large closed</Tag>
					<Tag style={{width: 200}}>overflow controls how a children are measured and displayed. overflow: hidden causes views to be clipped while overflow: scroll causes views to be measured independently of their parents main axis</Tag>

				</View>
				<View style={style.header}>
					<Text>横向布局</Text>
				</View>
				<View style={style.horizonContainer}>
					<Tag>默认tag</Tag>
					<Tag disabled>disabled</Tag>
					<Tag size="small">small</Tag>
				</View>
				<View style={style.header}>
					<Text>等分</Text>
				</View>
				<View style={style.horizonContainer}>
					<Tag style={{flex: 1}}>标签1</Tag>
					<Tag style={{flex: 1}} disabled>disabled</Tag>
					<Tag style={{flex: 2}} size="small">small</Tag>
				</View>
			</ScrollView>
		)
	}
}

