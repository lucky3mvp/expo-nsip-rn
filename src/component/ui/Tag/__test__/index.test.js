import React from 'react'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Tag from '../index'

describe('Tag Tests', () => {
	it('renders correctly', () => {
		const tree = Renderer.create((
			<Tag text='这是一个tag' />
		)).toJSON();
		
		expect(tree).toMatchSnapshot();
	});

	it('defaultSelected state', () => {
		let wrapper = shallow(<Tag>I’m a tag</Tag>);
		expect(wrapper.instance().state.selected).toBe(false);

		wrapper.simulate('press');
		expect(wrapper.instance().state.selected).toBe(true);
	});

	it('press event', () => {
		let handler = jest.fn();
		let wrapper = shallow(<Tag onPress={handler}>I’m a tag too</Tag>);
		
		expect(wrapper.instance().state.selected).toBe(false);
		wrapper.simulate('press');
		expect(wrapper.instance().state.selected).toBe(true);
	});
})
