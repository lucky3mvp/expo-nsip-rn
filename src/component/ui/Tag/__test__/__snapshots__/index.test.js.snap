// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Tests renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderColor": "#bbb",
        "borderRadius": 3,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 5,
        "paddingVertical": 5,
      },
      Object {},
    ]
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#888",
          "fontSize": 12,
        },
        Object {},
        Object {},
      ]
    }
  >
    这是一个tag
  </Text>
</View>
`;
